src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java

```diff
@@ -106,57 +105,32 @@ private String quoteValueIfRequired(String propertyValue) {
         return propertyValue;
     }
 
-    /**
-     * Returns {@code true} when metric name is a labeled metric name.
-     */
-    private boolean hasLabel(String name) {
-        return name.lastIndexOf(']') == name.length() - 1
-                && name.indexOf('[') >= 1
-                && name.indexOf('[') == name.lastIndexOf('[')
-                && name.indexOf(']') == name.lastIndexOf(']');
-    }
-
     /**
      * Extracts the name from a labeled metric name.
      */
     private String extractMetricName(String labeledMetricName) {
-        if (!hasLabel(labeledMetricName)) {
+        if (!LabeledMetric.hasLabel(labeledMetricName)) {
             return labeledMetricName;
         }
         return labeledMetricName.substring(0, labeledMetricName.indexOf("["));
     }
 
     /**
-     * Extracts the labels from a labeled metric name. Note: Spaces are significant everywhere in an Object Name. Do not
+     * Extract and quotes values of label if required. If metricName is not a labeled metric, the returned value
+     * has a leading ',' for easier concatenation to other parts of ObjectName.
+     *
+     * Note: Spaces are significant everywhere in an Object Name. Do not
      * write "metric_name[type=Thread, name=DGC] (with a space after the comma) because it will be interpreted as having
      * a key called " name", with a leading space in the name.
      */
-    private String extractLabels(String labeledMetricName) {
-        if (!hasLabel(labeledMetricName)) {
+    private String extractQuotedLabelValues(String metricName) {
+        if (!LabeledMetric.hasLabel(metricName)) {
             return "";
         }
 
-        return labeledMetricName.substring(
-                labeledMetricName.indexOf("[") + 1, labeledMetricName.lastIndexOf("]"));
-    }
-
-    /**
-     * Quotes values of label if required. If metricLabels is not blank, the returned value has a leading ',' for easier
-     * concatenation to other parts of ObjectName
-     */
-    private String quoteLabelValues(String metricLabels) {
-        if (metricLabels == null || metricLabels.length() == 0) {
-            return "";
-        }
-        StringBuilder labelBuilder = new StringBuilder(metricLabels.length()+1);
-        for (String label : metricLabels.split(",")) {
-            String[] labelParts = label.split("=");
-            if (labelParts.length != 2) {
-                throw new AssertionError("Illegal label provided: " + label);
-            }
-            labelBuilder.append(',');
-            labelBuilder.append(labelParts[0]).append('=').append(quoteValueIfRequired(labelParts[1]));
-        }
+        StringBuilder labelBuilder = new StringBuilder(metricName.length()+1);
```

Comment:
```suggestion
        StringBuilder labelBuilder = new StringBuilder(metricName.length() + 1);
```

Spaces before and after operator!